ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutor.scala

```diff
@@ -62,6 +65,9 @@ private[apiserver] final class LedgerTimeAwareCommandExecutor(
           val usedContractIds: Set[ContractId] = cer.transaction
             .inputContracts[ContractId]
             .collect { case id: ContractId => id }
+          val usedDisclosedContracts = cer.submitterInfo.explicitDisclosure.toSeq.toSet
```

Comment:
Note: avoid transitioning through set if not needed - hashCode computation can be very expensive for contracts arguments.

Reply:
Good point, we only use `usedDisclosedContracts` to find their maximum LET. We can move the `.toSet` to `usedDisclosedContractIds` (which is only a set of contract ids) below, and can replace the whole `usedDisclosedContracts` by just the maximum LET.